cmd/contour/serve.go

```diff
@@ -215,6 +215,24 @@ func NewServer(log logrus.FieldLogger, ctx *serveContext) (*Server, error) {
 		Scheme:                 scheme,
 		MetricsBindAddress:     "0",
 		HealthProbeBindAddress: "0",
+		NewCache: ctrl_cache.BuilderWithOptions(ctrl_cache.Options{
+			TransformByObject: ctrl_cache.TransformByObject{
+				&corev1.Secret{}: func(obj interface{}) (interface{}, error) {
+					secret, _ := obj.(*corev1.Secret)
+					if secret.Type == corev1.SecretTypeTLS || secret.Type == corev1.SecretTypeOpaque {
+						return obj, nil
+					}
+					secret.Data = nil
+					// Returning error will avoid handlers from being called.
+					// Object will still be cached.
+					return nil, fmt.Errorf("ignoring secret %s/%s of type %s", secret.Namespace, secret.Name, secret.Type)
+				},
+			},
+			DefaultTransform: func(obj interface{}) (interface{}, error) {
+				obj.(client.Object).SetManagedFields(nil)
+				return obj, nil
+			},
```

Comment:
Based on godoc it seems like this won't get applied to Secrets, so would be good to clear ManagedFields for all of them as well